### PR TITLE
Added preference for the instantiation results dialog

### LINKED ISSUE #2274: Instantiation results dialog is annoying when instantiation is successful
The instantiation results dialog always shows after completion, but it's annoying to make it go away each time.

There should be a workspace preference to only show it when there are unsuccessful instantiations.

### PR DESCRIPTION
**Pull request by AaronGreenhouse**
Fixed #2274

### CODE REVIEW

**Comment by AaronGreenhouse** [comment]
Need to add an option to the result dialog box itself to make it go away.

Also, for this and for the Flow Latency dialog, add a secondary dialog that explains how to go to the preferences and get it back.

File: analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/dialogs/FlowLatencyDialog.java [hunk 1]
```diff
@@ -184,16 +185,23 @@ protected void okPressed() {
...
+		}
 
+		savePreferences();
 		super.okPressed();
 	}
 
 	private void savePreferences() {
 		if (latencyPrefs.needsSaving()) {
-			final Job saveJob = Job.create("Save latency prefencees", monitor -> {
+			final Job saveJob = Job.create("Save latency prefences", monitor -> {
```
Comment (lwrage) [inline analyses/org.osate.analysis.flows/src/org/osate/analysis/flows/dialogs/FlowLatencyDialog.java:204]: typo

File: core/org.osate.core/src/org/osate/core/preferences/PreferenceInitializer.java [hunk 2]
```diff
@@ -48,6 +48,7 @@ public void initializeDefaultPreferences() {
 		IPreferenceStore store = OsateCorePlugin.getDefault().getPreferenceStore();
 		store.setDefault(OsateCorePlugin.EXPAND_DEFAULT_FLAG, false);
 		store.setDefault(OsateCorePlugin.MAX_SOM, OsateCorePlugin.MAX_SOM_DEFAULT);
+		OsateCorePlugin.setAlwaysShowInstantiationResultsDefault(store);
```
Comment (lwrage) [inline core/org.osate.core/src/org/osate/core/preferences/PreferenceInitializer.java:51]: Why is this done via a method, different from the other default settings?

File: core/org.osate.core/src/org/osate/core/OsateCorePlugin.java [hunk 3]
```diff
@@ -72,6 +72,11 @@
 	public static final String MAX_SOM = "maxSOM";
 	public static final int MAX_SOM_DEFAULT = 1000;
 
+	// Package visible: only used locally any how, and this way we don't break the plug-in versioning
+	static final String ALWAYS_SHOW_INSTANTIATION_RESULTS = "alwaysShowInstantiationResults";
+	// Package visible: only used locally any how, and this way we don't break the plug-in versioning
+	static final boolean ALWAYS_SHOW_INSTANTIATION_RESULTS_DEFAULT = true;
```
Comment (lwrage) [inline core/org.osate.core/src/org/osate/core/OsateCorePlugin.java:78]: Do we really need a constant for this?

File: core/org.osate.core/src/org/osate/core/OsateCorePlugin.java [hunk 4]
```diff
@@ -339,4 +344,27 @@ public final int getSOMLimit() {
...
+		store.setDefault(OsateCorePlugin.ALWAYS_SHOW_INSTANTIATION_RESULTS,
+				OsateCorePlugin.ALWAYS_SHOW_INSTANTIATION_RESULTS_DEFAULT);
+	}
+
+	/**
+	 * @since 2.1
+	 */
+	public static String getAlwaysShowInstantiationResultsPreferenceName() {
+		return ALWAYS_SHOW_INSTANTIATION_RESULTS;
+	}
```
Comment (lwrage) [inline core/org.osate.core/src/org/osate/core/OsateCorePlugin.java:369]: I'd prefer to not have these two methods. IMO, they just make code more difficult to follow.

**Comment by AaronGreenhouse** [comment]
Also closed #2294

File: core/org.osate.help/help/markdown/instantiation.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+This program includes and/or can make use of certain third party source code, object code, documentation and other
+files ("Third Party Software"). The Third Party Software that is used by this program is dependent upon your system
+configuration. By using this program, You agree to comply with any and all relevant Third Party Software terms and
+conditions contained in any such Third Party Software or separate license file distributed with such Third Party
+Software. The parties who own the Third Party Software ("Third Party Licensors") are intended third party benefici-
+aries to this license with respect to the terms applicable to their Third Party Software. Third Party Software li-
+censes only apply to the Third Party Software and not any other portion of this program or this program as a whole.
+-->
+# System Instantiation
```
Comment (lwrage) [inline core/org.osate.help/help/markdown/instantiation.md:24]: Please review for typos /grammar.

**Pull request merged**